### ISSUE TITLE
Motor Action (Up, Down or Stop) in Home Assistant throw Exceptions

### ISSUE THREAD

**Issue by LtPixel** [ridiekel/jeletask#37]
I've added my curtain motor into the Jeletask configuration... * It's defined as a "roller blind" in Teletask.

**jeletask.conf**

```
"MOTOR": [
      {
        "number": 1,
        "description": "Living Curtains"
      }
]
```

Upon launching an action in Home Assistant (up, down or stop), I receive the following message in the Jeletask console...

```
2023-09-01 12:20:42.854 - INFO - i.g.r.j.mqtt.listener.MqttProcessor      : [COMMAND] - [MOTOR     ] - [  4] - [                Living Curtains] - {"state": "DOWN"}
2023-09-01 12:20:42.866 - INFO - i.g.r.j.client.TeletaskClientImpl        : State for MOTOR / 4 was somehow null, we reset the state to: null
2023-09-01 12:20:42.880 -ERROR - i.g.r.j.c.b.m.messages.MessageSupport    : Exception (java.lang.ArrayIndexOutOfBoundsException) caught in extractMessages: Index 3 out of bounds for length 3

java.lang.ArrayIndexOutOfBoundsException: Index 3 out of bounds for length 3
        at io.github.ridiekel.jeletask.client.builder.composer.config.statecalculator.MotorStateCalculator.toComponentState(MotorStateCalculator.java:42) ~[jeletask-client-4.3.3.jar!/:4.3.3]
        at io.github.ridiekel.jeletask.client.builder.composer.config.ConfigurationSupport.getState(ConfigurationSupport.java:38) ~[jeletask-client-4.3.3.jar!/:4.3.3]
        at io.github.ridiekel.jeletask.client.builder.composer.microsplus.MicrosPlusCommandConfiguration$MicrosPlusEventCommandConfigurable.parse(MicrosPlusCommandConfiguration.java:50) ~[jeletask-client-4.3.3.jar!/:4.3.3]
        at io.github.ridiekel.jeletask.client.builder.composer.microsplus.MicrosPlusCommandConfiguration$MicrosPlusEventCommandConfigurable.parse(MicrosPlusCommandConfiguration.java:41) ~[jeletask-client-4.3.3.jar!/:4.3.3]
        at io.github.ridiekel.jeletask.client.builder.composer.MessageHandlerSupport.parse(MessageHandlerSupport.java:74) ~[jeletask-client-4.3.3.jar!/:4.3.3]
        at io.github.ridiekel.jeletask.client.builder.message.MessageUtilities.extractMessages(MessageUtilities.java:76) ~[jeletask-client-4.3.3.jar!/:4.3.3]
        at io.github.ridiekel.jeletask.client.builder.message.MessageUtilities.receive(MessageUtilities.java:41) ~[jeletask-client-4.3.3.jar!/:4.3.3]
        at io.github.ridiekel.jeletask.client.builder.message.messages.MessageSupport.lambda$waitForAcknowledge$0(MessageSupport.java:77) ~[jeletask-client-4.3.3.jar!/:4.3.3]
        at org.awaitility.core.CallableCondition$ConditionEvaluationWrapper.eval(CallableCondition.java:99) ~[awaitility-4.0.1.jar!/:na]
        at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:203) ~[awaitility-4.0.1.jar!/:na]
        at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:190) ~[awaitility-4.0.1.jar!/:na]
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[na:na]
        at org.awaitility.core.SameThreadExecutorService.execute(SameThreadExecutorService.java:79) ~[awaitility-4.0.1.jar!/:na]
        at java.base/java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:145) ~[na:na]
        at org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:92) ~[awaitility-4.0.1.jar!/:na]
        at org.awaitility.core.CallableCondition.await(CallableCondition.java:78) ~[awaitility-4.0.1.jar!/:na]
        at org.awaitility.core.CallableCondition.await(CallableCondition.java:26) ~[awaitility-4.0.1.jar!/:na]
        at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:873) ~[awaitility-4.0.1.jar!/:na]
        at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:842) ~[awaitility-4.0.1.jar!/:na]
        at io.github.ridiekel.jeletask.client.builder.message.messages.MessageSupport.waitForAcknowledge(MessageSupport.java:75) ~[jeletask-client-4.3.3.jar!/:4.3.3]
        at io.github.ridiekel.jeletask.client.builder.message.messages.MessageSupport.execute(MessageSupport.java:58) ~[jeletask-client-4.3.3.jar!/:4.3.3]
        at io.github.ridiekel.jeletask.client.builder.message.executor.MessageExecutor.run(MessageExecutor.java:22) ~[jeletask-client-4.3.3.jar!/:4.3.3]
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[na:na]
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[na:na]
        at java.base/java.lang.Thread.run(Thread.java:833) ~[na:na]

2023-09-01 12:20:47.897 - WARN - i.g.r.j.mqtt.listener.MqttProcessor      : MQTT topic 'teletask2mqtt/micros_plus/motor/4/set' could not change state to: {"state": "DOWN"} -- Condition with alias 'State update' didn't complete within 5 seconds because condition with io.github.ridiekel.jeletask.client.TeletaskClientImpl was not fulfilled.

org.awaitility.core.ConditionTimeoutException: Condition with alias 'State update' didn't complete within 5 seconds because condition with io.github.ridiekel.jeletask.client.TeletaskClientImpl was not fulfilled.
        at org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:148) ~[awaitility-4.0.1.jar!/:na]
        at org.awaitility.core.CallableCondition.await(CallableCondition.java:78) ~[awaitility-4.0.1.jar!/:na]
        at org.awaitility.core.CallableCondition.await(CallableCondition.java:26) ~[awaitility-4.0.1.jar!/:na]
        at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:873) ~[awaitility-4.0.1.jar!/:na]
        at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:842) ~[awaitility-4.0.1.jar!/:na]
        at io.github.ridiekel.jeletask.client.TeletaskClientImpl.set(TeletaskClientImpl.java:88) ~[jeletask-client-4.3.3.jar!/:4.3.3]
        at io.github.ridiekel.jeletask.client.TeletaskClientImpl.set(TeletaskClientImpl.java:103) ~[jeletask-client-4.3.3.jar!/:4.3.3]
        at io.github.ridiekel.jeletask.mqtt.listener.MqttProcessor$ChangeStateMqttCallback.messageArrived(MqttProcessor.java:372) ~[jeletask2mqtt-4.3.3.jar!/:4.3.3]
        at org.eclipse.paho.client.mqttv3.internal.CommsCallback.deliverMessage(CommsCallback.java:511) ~[org.eclipse.paho.client.mqttv3-1.2.5.jar!/:na]
        at org.eclipse.paho.client.mqttv3.internal.CommsCallback.handleMessage(CommsCallback.java:417) ~[org.eclipse.paho.client.mqttv3-1.2.5.jar!/:na]
        at org.eclipse.paho.client.mqttv3.internal.CommsCallback.run(CommsCallback.java:214) ~[org.eclipse.paho.client.mqttv3-1.2.5.jar!/:na]
        at java.base/java.lang.Thread.run(Thread.java:833) ~[na:na]
```

Any idea why I'm receiving this exception? _* I perfectly able to manipulate the curtains using the Teletask app._

P.

**Comment by LtPixel** [comment]
I found the issue. 👍

In Teletask, under the configuration of the motor, I didn't set the "Advanced" option "Position indication". This was causing the exception (as Jeletask was probably trying to read the position of the curtain). 😉

I've now defined this as "Indication for all positions", and under the main "Settings" accurately defined the run time (from 0% to 100%, and from 100% to 0%). ✅

This solved the issue. 🚀
